docs/src/architecture/notes.md

```diff
@@ -1,73 +1,193 @@
-
-Privately stored notes can only be consumed if the note data is known to the consumer. That means, there must be some offchain communication to transmit the note's data from the sender to the receipient.
-
-# Note metadata
-For every note the Miden Operator stores metadata in the Note DB. This metadata includes:
-
-* A **user-defined tag** as a means to quickly grab all notes for a certain application or use case.
-* A **sender** to be able to provide also ERC20 contract functionality.
-* The **number of assets** contained in the note. 
-
-# Note hash
-The note hash is computed as:
+ In addition, a note has **metadata** including the sender and the note tag.
 
-`hash(hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash), vault_hash)`
+# Note's lifecycle
+New notes are created by executing transactions. After verifying the transaction proof the operator adds either only the note hash (private notes) or the full note data (public notes) to the [Note DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#notes-database). Notes can be produced and consumed locally by users in local transactions or by the operator in a network transaction. Note consumption requires the transacting party to know the note data to compute the nullifier. After successful verification, the operator sets the corresponding entry in the Nullifier DB to `1` (=consumed). 
 
-This achieves the following properties:
-- Every note can be reduced to a single unique hash.
-- To compute a note's hash, we do not need to know the note's `serial_num`. Knowing the hash
-    of the `serial_num` (as well as `script_hash`, `input_hash` and `note_vault`) is sufficient.
-- Moreover, the note hash can be computed having the [recipient](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html#note-recipient) and note vault, as `hash(recipient, vault_hash)`.
-- We compute the hash of `serial_num` as `hash(serial_num, [0; 4])` to simplify processing within the VM.
-
-# Note nullifier
-The nullifier is the note's index in the [Nullifier DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#nullifier-database). The Nullifier DB stores the information whether a note was already consumed.
+<p align="center">
+    <img src="../diagrams/architecture/note/Note_life_cycle.png">
+</p>
 
-The nullifier is computed as `hash(serial_num, script_hash, input_hash, vault_hash)`.
+The following sections will explain, how notes are created, stored, discovered and consumed. 
+
+## Note creation
+Notes are created as outputs (`OutputNotes`) of Miden transactions. Operators record those notes to the [Note DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#note-database), after successful verification of the underlying transactions. Then, those notes exist and can be consumed.
```

Comment:
Would change to: after successful verification of the underlying transactions those notes can be consumed.